open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/retrieval/flow/VRPConsentRetrievalUtilTest.java

```diff
@@ -0,0 +1,389 @@
+        Mockito.doReturn(ConsentAuthorizeTestConstants.COF_RECEIPT).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray cofConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(cofConsentData);
+    }
+
+    @Test
+    public void testGetConsentDataSetForVRPData() {
+
+        Mockito.doReturn(ConsentExtensionConstants.VRP).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.VRP_INITIATION).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray vrpConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(vrpConsentData);
+    }
+
+    @Test
+    public void testAppendDummyAccountID() {
+
+        JSONArray result = ConsentRetrievalUtil.appendDummyAccountID();
+
+        Assert.assertNotNull(result);
+
+        Assert.assertEquals(result.size(), 2);
+
+        JSONObject accountOne = (JSONObject) result.get(0);
+        Assert.assertEquals(accountOne.get("account_id"), "12345");
+        Assert.assertEquals(accountOne.get("display_name"), "Salary Saver Account");
+
+        JSONObject accountTwo = (JSONObject) result.get(1);
+        Assert.assertEquals(accountTwo.get("account_id"), "67890");
+        Assert.assertEquals(accountTwo.get("display_name"), "Max Bonus Account");
+    }
+
+    @Test(expectedExceptions = ConsentException.class)
+    public void testConsentDataWithInvalidJson() {
+        // Arrange
+        String invalidJsonString = "Invalid JSON String";
+        Mockito.when(consentResourceMock.getReceipt()).thenReturn(invalidJsonString);
+
+        // Act
+        JSONArray consentDataJSON = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+        Assert.assertNotNull(consentDataJSON);
+    }
+
+    @Test(expectedExceptions = ConsentException.class)
+    public void testGetConsentDataSetForVRPDataParameter() {
+
+        Mockito.doReturn(ConsentExtensionConstants.VRP).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(VRP_WITHOUT_DATA).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray vrpConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(vrpConsentData);
+
+
+        Assert.assertNotNull(ConsentExtensionConstants.IS_ERROR);
+        String errorMsg = ConsentExtensionConstants.IS_ERROR;
+        Assert.assertTrue(errorMsg.contains(ErrorConstants.NOT_JSON_PAYLOAD));
```

Comment:
Here also we are assigning `errorMsg` as `ConsentExtensionConstants.IS_ERROR` , and then whether it contains `ErrorConstants.NOT_JSON_PAYLOAD` , this logic seems not clear, could you explain?

Reply:
String errorMsg = ConsentExtensionConstants.IS_ERROR,  this code retrieves the actual error message .
Assert.assertTrue(errorMsg.contains(ErrorConstants.NOT_JSON_PAYLOAD)); ,this code checks whether the error message contains the specific string defined in ErrorConstants.NOT_JSON_PAYLOAD. If the substring is found within the error message, the assertion passes.

Fixed in [4e69857]